crates/services/relayer/src/service/syncing.rs

```diff
@@ -46,10 +46,13 @@ where
 {
     match eth_node.syncing().await {
         Ok(s) => Ok(s),
-        Err(err) => Err(anyhow::anyhow!(
-            "Failed to check if DA layer is syncing {}",
-            err
-        )),
+        Err(err) => {
+            tracing::error!(?err, "failed to check if DA layer is syncing");
+            Err(anyhow::anyhow!(
+                "Failed to check if DA layer is syncing {}",
+                err
+            ))
```

Comment:
We can probably throw this all into the error, since the whole error will get logged by the service runner.

This works too though.

Reply:
Yeah, I was a bit torn about that. It feels so wrong to put debug information within the actual error, since the canonical display representation is already provided. But, on the other hand it is practical to get the information there because it's currently missing.

However, the deciding factor for me is that the log message is structured, so it's much easier to parse and hide bloated fields in log explorers (although we need to improve ours). But if we bloat the actual message, it ends up polluting the logs a bit more and is more tedious to clean out because typically you'll want to render the message.

I can go either way here though, and need to sleep soon so feel free to update the PR and add this to the log as well.

Reply:
I also prefer to just include this information into the error, instead of adding additional log for the same event

Reply:
Fair enough, I've updated to include the debug information in the error instead here https://github.com/FuelLabs/fuel-core/pull/3004/commits/7a52b5039f6ad02b18db4ca30bec4f5d61726424